blockchain/rolling_merkle.go

```diff
@@ -0,0 +1,149 @@
+package blockchain
+
+import (
+	"crypto/sha256"
+	"math/bits"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+)
+
+// parentHash returns the hash of the left and right hashes passed in.
+func parentHash(l, r chainhash.Hash) chainhash.Hash {
```

Comment:
So this is the same as `HashMerkleBranches`?

Reply:
Oh yeah you're right, it serves the same functionality.

Though the `parentHash` function is faster because there's no allocation being done vs `HashMerkleBranches` which does a single allocation. I've tested and verified that this is because `HashMerkleBranches` is returning a pointer.

Would it be ok if I were to change the `HashMerkleBranches` function to return a `chainhash.Hash` instead of a pointer? Then `parentHash` and `HashMerkleBranches` would be equal.

```
BenchmarkHashMerkleBranches-10           2279146               526.5 ns/op            32 B/op          1 allocs/op
BenchmarkParentHash-10                   2392340               497.1 ns/op             0 B/op          0 allocs/op
```

I'll modify the 

Code used for the above benchmark:

```
func BenchmarkHashMerkleBranches(b *testing.B) {
       var aHash, bHash chainhash.Hash
       for i := 0; i < b.N; i++ {
               HashMerkleBranches(&aHash, &bHash)
       }
}

func BenchmarkParentHash(b *testing.B) {       
       var aHash, bHash chainhash.Hash
       for i := 0; i < b.N; i++ {
               parentHash(aHash, bHash)
       }
}
```

Reply:
Added a commit to change the `HashMerkleBranches` to return a `chainhash.Hash`. Removed `parentHash`.